src/main/ApplicationImpl.cpp

```diff
@@ -215,11 +215,17 @@ maybeRebuildLedger(Application& app, bool applyBuckets)
         // this will run again on next start up.
         if (applyBuckets)
         {
-            LOG_INFO(DEFAULT_LOG,
-                     "Rebuilding ledger tables by applying buckets");
-            if (!applyBucketsForLCL(app))
+            // If we are still on the genesis ledger, we don't need to apply
+            // buckets
+            auto has = app.getLedgerManager().getLastClosedLedgerHAS();
+            if (has.currentLedger != 0)
```

Comment:
genesis ledger is 1, so why are we checking for 0 here? currentLedger=0 is not a valid HAS, so I'm wondering if the bug is somewhere else, and this is just a symptom?

Reply:
A default constructed HAS has a currentLedger == 0. The issue here is that if we crash/exit very early on startup, we still set the offer rebuild flag but haven;t persisted a HAS yet. After new-db, we don't set the rebuild flag, so we don't attempt to apply a "null" has. However if we crash before generating a non-null has, we attempt to assume state against the invalid HAS. Really what should be happening here is that `getLastClosedLedgerHAS` should return an optional that's null until we've populated the genesis bucket and have a valid has. This is kinda a lot of refactoring though and seems like an edge case that only happens at the very start of startup.